src/common/options/rgw.yaml.in

```diff
@@ -277,6 +277,21 @@ options:
   see_also:
   - rgw_cache_enabled
   with_legacy: true
+- name: rgw_cache_shard_count
+  type: uint 
+  level: advanced
+  desc: The number of RGW cache shards.
+  long_desc: The number of independent cache shards to use for the rados object gateway.
+    Each cache shard will store up to rgw_cache_lru_size items.
+  fmt_desc: The number of Coph Object Gateway cache shards.
+  default: 4
```

Comment:
myb consider prime like `5` as default

Reply:
@markhpc possibly a rebase is needed, encountered compilation errors when cherry-picking commit 7f3d483883d4ac2ada4ed93284eb06b2429666f0 over master

Reply:
> @mkogan1 ok, I'm swamped right now but will try to get back to it sooner or later! Do you still think we should break the mutex part off into a separate PR? I figured they sort of worked together given that the sharding should reduce contention on the lock anyway (making the lock/unlock overhead a more substantial portion of the overall overhead), but I'm open.

@markhpc think better to split the mutex commit as it seems that it can be merged while the sharding still requires addressing reviewer comments
if no objection I can pick up the work on this and address the comments including working on switching to intrusive list as suggested

Reply:
@mkogan1 sure, go for it!  You know this code better than I do anyway.